chemprop/v2/data/splitting.py

```diff
@@ -206,9 +192,54 @@ def _unpack_astartes_result(
     else:
         train_idxs, test_idxs = result[-2], result[-1]
-    if data is None:
-        return train_idxs, val_idxs, test_idxs
-    train = [data[i] for i in train_idxs]
-    val = [data[i] for i in val_idxs]
-    test = [data[i] for i in test_idxs]
+    return list(train_idxs), list(val_idxs), list(test_idxs)
+
+
+def split_monocomponent(
+    datapoints: Sequence[MoleculeDatapoint], split: SplitType | str = "random", **kwargs
+) -> tuple[list[MoleculeDatapoint], ...] | tuple[list[list[MoleculeDatapoint]], ...]:
+    """Splits monocomponent data into training, validation, and test splits."""
+
+    # split the data
+    train_idxs, val_idxs, test_idxs = split_data(datapoints, **kwargs)
+
+    match SplitType.get(split):
+        case SplitType.CV_NO_VAL, SplitType.CV:
+            # convert indices to datapoints for each fold
+            train = [[datapoints[i] for i in fold] for fold in train_idxs]
+            val = [[datapoints[i] for i in fold] for fold in val_idxs]
+            test = [[datapoints[i] for i in fold] for fold in test_idxs]
+        case SplitType.SCAFFOLD_BALANCED, SplitType.RANDOM_WITH_REPEATED_SMILES, SplitType.RANDOM, SplitType.KENNARD_STONE, SplitType.KMEANS:
+            # convert indices to datapoints
+            train = [datapoints[i] for i in train_idxs]
+            val = [datapoints[i] for i in val_idxs]
+            test = [datapoints[i] for i in test_idxs]
+        case _:
+            raise ValueError(f'Split type "{split}" not supported.')
+
+    return train, val, test
+
+
+def split_multicomponent(
+    datapointss: Sequence[MulticomponentDatapoint], split: SplitType | str = "random", key_index: int = 0, **kwargs
+) -> tuple[list[MulticomponentDatapoint], ...] | tuple[list[list[MulticomponentDatapoint]], ...]:
+    """Splits multicomponent data into training, validation, and test splits."""
+
+    key_datapoints = datapointss[key_index]
+    train_idxs, val_idxs, test_idxs = split_data(key_datapoints, **kwargs)
+
+    match SplitType.get(split):
+        case SplitType.CV_NO_VAL, SplitType.CV:
+            # convert indices to datapoints for each fold
+            train = [[datapoints[i] for i in fold] for datapoints in datapointss for fold in train_idxs]
+            val = [[datapoints[i] for i in fold] for datapoints in datapointss for fold in val_idxs]
+            test = [[datapoints[i] for i in fold] for datapoints in datapointss for fold in test_idxs]
+        case SplitType.SCAFFOLD_BALANCED, SplitType.RANDOM_WITH_REPEATED_SMILES, SplitType.RANDOM, SplitType.KENNARD_STONE, SplitType.KMEANS:
+            # convert indices to datapoints
+            train = [[datapoints[i] for i in train_idxs] for datapoints in datapointss]
+            val = [[datapoints[i] for i in val_idxs] for datapoints in datapointss]
+            test = [[datapoints[i] for i in test_idxs] for datapoints in datapointss]
+        case _:
+            raise ValueError(f'Split type "{split}" not supported.')
```

Comment:
this should be a 
```suggestion
            raise RuntimeError("Unreachable code reached!")
```
the above call to `SplitType.get` should raise the appropriate `ValueError` and it should never fall through to this case, but it's always good practice to make sure that it doesn't.

Reply:
Thanks! I committed this and also change the other similar exceptions.